kv/kvclient/kvtenant: TestRangeLookupPrefetchFiltering failed

kv/kvclient/kvtenant.TestRangeLookupPrefetchFiltering [failed](https://tanzanite.cluster.engflow.com/invocations/default/ba674aaf-7120-40ec-8b10-13420f95de27?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZjbGllbnQva3Z0ZW5hbnQ6a3Z0ZW5hbnRfdGVzdA==) on master @ [7cea6a90eed456c76dfa07e618c3cd2257b302e5](https://github.com/cockroachdb/cockroach/commits/7cea6a90eed456c76dfa07e618c3cd2257b302e5):


```
=== RUN   TestRangeLookupPrefetchFiltering
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestRangeLookupPrefetchFiltering2086136343
    test_log_scope.go:81: use -show-logs to present logs inline
    tenant_range_lookup_test.go:82: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvtenant_test/pkg/kv/kvclient/kvtenant/tenant_range_lookup_test.go:82
        	Error:      	"[r70:/Tenant/2{a-b} [(n1,s1):1, next=2, gen=5, sticky=9223372036.854775807,2147483647] r74:/Tenant/2{b-/Table/11} [(n1,s1):1, next=2, gen=5, sticky=9223372036.854775807,2147483647] r71:/Tenant/2/Table/1{1-2} [(n1,s1):1, next=2, gen=5] r73:/Tenant/2/Table/{12-25} [(n1,s1):1, next=2, gen=6] r75:/Tenant/2/Table/2{5-6} [(n1,s1):1, next=2, gen=7] r76:/Tenant/2/Table/2{6-7} [(n1,s1):1, next=2, gen=8] r77:/Tenant/2/Table/2{7-8} [(n1,s1):1, next=2, gen=9] r78:/Tenant/{2/Table/28-3} [(n1,s1):1, next=2, gen=9]]" should have 2 item(s), but has 8
        	Test:       	TestRangeLookupPrefetchFiltering
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestRangeLookupPrefetchFiltering2086136343
--- FAIL: TestRangeLookupPrefetchFiltering (14.15s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRangeLookupPrefetchFiltering.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-37326

Reproduces on the same SHA as the test failure 7cea6a90eed456c76dfa07e618c3cd2257b302e5:

```
I240402 14:57:11.021663 1 (gostd) rand.go:250  [-] 1  random seed: -1657496049575980794
    test_log_scope.go:170: test logs captured to: /home/kvoli/go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/_tmp/7219c67f95200c95a7445ed368be8564/logTestRangeLookupPrefetchFiltering1329605097
    test_log_scope.go:81: use -show-logs to present logs inline
    tenant_range_lookup_test.go:82:
                Error Trace:    github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvtenant_test/pkg/kv/kvclient/kvtenant/tenant_range_lookup_test.go:82
                Error:          "[r70:/Tenant/2{a-b} [(n1,s1):1, next=2, gen=4, sticky=9223372036.854775807,2147483647] r71:/Tenant/2{b-/Table/11} [(n1,s1):1, next=2, gen=5, sticky=9223372036.854775807,2147483647] r73:/Tenant/{2/Table/11-3} [(n1,s1):1, next=2, gen=5]]" should have 2 item(s), but has 3
                Test:           TestRangeLookupPrefetchFiltering
    panic.go:626: -- test log scope end --
test logs left over in: /home/kvoli/go/src/github.com/cockroachdb/cockroach/artifacts/_tmp/_tmp/7219c67f95200c95a7445ed368be8564/logTestRangeLookupPrefetchFiltering1329605097
--- FAIL: TestRangeLookupPrefetchFiltering (20.35s)
```

Closing as fixed.

--- a/pkg/kv/kvclient/kvtenant/tenant_range_lookup_test.go
+++ b/pkg/kv/kvclient/kvtenant/tenant_range_lookup_test.go
@@ -18,7 +18,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/keys"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangecache"
-	"github.com/cockroachdb/cockroach/pkg/kv/kvserver"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
@@ -34,13 +33,11 @@ func TestRangeLookupPrefetchFiltering(t *testing.T) {
 
 	ctx := context.Background()
 	tc := testcluster.StartTestCluster(t, 1, base.TestClusterArgs{
+		// Avovid additional splits (outside those below) by disabling the split
+		// queue via the replication manual mode.
+		ReplicationMode: base.ReplicationManual,
 		ServerArgs: base.TestServerArgs{
 			DefaultTestTenant: base.TestControlsTenantsExplicitly,
-			Knobs: base.TestingKnobs{
-				Store: &kvserver.StoreTestingKnobs{
-					DisableMergeQueue: true,
-				},
-			},
 		},
 	})
 	defer tc.Stopper().Stop(ctx)

